File: jinia/object/src/main/java/chapter2/model/Screening.java
package chapter2.model;

import java.time.ZonedDateTime;

public class Screening {

    private Long id;
    private Movie movie;
    private Integer sequence;
    private ZonedDateTime whenScreened;

    public Screening(Long id, Movie movie, Integer sequence, ZonedDateTime whenScreened) {
        this.id = id;
        this.movie = movie;
        this.sequence = sequence;
        this.whenScreened = whenScreened;
    }

    public Long getId() {
        return id;
    }

    public Money getMovieFee() {
        return movie.getFee();
    }

    public boolean isSequence(Integer sequence) {
        return this.sequence.equals(sequence);
    }

    public ZonedDateTime getStartTime() {
        return whenScreened;
    }

    public MovieScreeningBooking booking(Customer customer, int audienceCount){
        return new MovieScreeningBooking(customer, this,calculateFee(audienceCount), audienceCount);
    }

    private Money calculateFee(int audienceCount) {
        return movie.calculateMovieFee(this).times(audienceCount);
    }
}

File: prravda/src/theater-booking-application/screening.ts
import { CreateScreeningDto } from "./dto/create-screening-dto";
import { Movie } from "./movie";
import { Reservation } from "./reservation";

export class Screening {
  constructor(
    private movie: Movie,
    private sequence: number,
    private whenScreened: Date
  ) {}

  private calculateFee(audienceCount: number) {
    return this.movie.calculateMovieFee().times(audienceCount);
  }

  public reserve(customer: Customer, audienceCount: number) {
    return new Reservation(
      customer,
      this,
      this.calculateFee(audienceCount),
      audienceCount
    );
  }

  public getStartTime() {
    return this.whenScreened;
  }

  public isSequence(sequence: number) {
    return this.sequence === sequence;
  }

  public getMovieFee() {
    return this.movie.getFee();
  }
}
